### ISSUE TITLE
Index Flow terminates and closes, but keeps showing activity

### ISSUE THREAD

**Issue by alexcg1** [jina-ai/jina#2343]
**Describe the bug**
I'm re-testing the Wikipedia example with alternative dataset. After hitting [this bug](https://github.com/jina-ai/examples/issues/474) I changed `flows/index.yml` to `restful: False` to stop indexing errors.

Now it seems to index okay, but it doesn't stop. Even after it says Flow is terminated, terminal output still keeps scrolling and the process remains active (see screenshots)

**Describe how you solve it**
- Hitting Ctrl-C doesn't work.
- I have to go in and kill-9 the process
- Running a query works just fine afterwards

---

**Environment**

```
- jina 1.1.0
- jina-proto 0.0.79
- jina-vcs-tag (unset)
- libzmq 4.3.4
- pyzmq 1.20.2
- protobuf 3.15.8
- proto-backend cpp
- grpcio 1.37.0
- pyyaml 5.4.1
- python 3.9.2
- platform Linux
- platform-release 5.10.26-1-MANJARO
- platform-version #1 SMP PREEMPT Thu Mar 25 16:56:17 UTC 2021
- architecture x86_64
- processor
- jina-resources /home/alexcg/tmp/examples/wikipedia-
  sentences/env/lib/python3.9/site-packages/jina/resources
* JINA_ARRAY_QUANT (unset)
* JINA_BINARY_DELIMITER (unset)
* JINA_CONTRIB_MODULE (unset)
* JINA_CONTRIB_MODULE_IS_LOADING(unset)
* JINA_CONTROL_PORT (unset)
* JINA_DEFAULT_HOST (unset)
* JINA_DISABLE_UVLOOP (unset)
* JINA_EXECUTOR_WORKDIR (unset)
* JINA_FULL_CLI (unset)
* JINA_IPC_SOCK_TMP (unset)
* JINA_LOG_CONFIG (unset)
* JINA_LOG_ID (unset)
* JINA_LOG_LEVEL (unset)
* JINA_LOG_NO_COLOR (unset)
* JINA_LOG_WORKSPACE (unset)
* JINA_POD_NAME (unset)
* JINA_RAISE_ERROR_EARLY (unset)
* JINA_RANDOM_PORTS (unset)
* JINA_RANDOM_PORT_MAX (unset)
* JINA_RANDOM_PORT_MIN (unset)
* JINA_SOCKET_HWM (unset)
* JINA_VCS_VERSION (unset)
* JINA_WARN_UNNAMED (unset)
* JINA_WORKSPACE (unset)
```
Note: Running `jina --version-full` also brings up following:

```
/home/alexcg/tmp/examples/wikipedia-sentences/env/lib/python3.9/site-packages/jina/types/document/__init__.py:66: RuntimeWarning: Module "tensorflow>=2.0" is not installed, fallback to default behavior. You are trying to use an extension feature not enabled by the current installation.
This feature is available in: [tensorflow>=2.0] [framework] [cicd]
Use pip install jina[TAG] to enable it
  tensorflow_installed = True
/home/alexcg/tmp/examples/wikipedia-sentences/env/lib/python3.9/site-packages/jina/importer.py:266: UserWarning: due to the missing dependencies or bad implementations, ['jina.hub.encoders.multimodal.TirgMultimodalEncoder', 'jina.hub.classifiers.image.CLIPZeroShotClassifier'] can not be imported if you are using these executors/drivers, they wont work. You can use `jina check` to list all executors and drivers
  warnings.warn(
```

Nothing comes after that opening bracket at the end which seems odd.

**Screenshots**
![Screenshot from 2021-04-23 12-16-38](https://user-images.githubusercontent.com/4182659/115857716-4ea7c200-a42e-11eb-81b2-8ebe9364f2d7.png)

![Screenshot from 2021-04-23 12-20-00](https://user-images.githubusercontent.com/4182659/115857692-4780b400-a42e-11eb-954b-d297336bfc71.png)

**Comment by JoanFM** [comment]
Hey @alexcg1 ,

can u have here a detailed description on the steps to reproduce the issue exaclty? What dataset is used? Which files and changes are u applying?

What exact commands are being run?

**Comment by JoanFM** [comment]
Hey @alexcg1 ,

This looks very weird. If you check the `PID` values from the `logs`, it shows that the Encoder Runtime runs in number: 157341.

However, the `encoder` receiving requests after `close` is called, has the number:  `154060`.

So my guess is:

In a previous run, u had killed the Flow using `Contrl+C` before the Flow finished. 
And afterwards, u run a regular Flow till the end, but since the past `Flow` did not close properly (as explained in #2352  #2353), you still have that `process` running.

So I would say the underlying problem is the `flushing` of extra data when running `kbinterrupt` and will be fixed in the scope of #2353  and #2352

**Comment by alexcg1** [comment]
Cool cool. During our screen share I couldn't replicate, so I'll just close this issue
